packages/plugin-core/src/_extension.ts

```diff
@@ -1243,3 +1248,67 @@ function updateEngineAPI(port: number | string): void {
   // const engine = ext.getEngine();
   // return engine;
 }
+
+function warnConflictingExtensions() {
+  const installedExtensions = CONFLICTING_EXTENSIONS.filter(
+    (conflictingExtension) => {
+      return VSCodeUtils.isExtensionInstalled(conflictingExtension);
+    }
+  );
+  const shouldDisplayWarning = installedExtensions.length > 0;
+  if (shouldDisplayWarning) {
+    AnalyticsUtils.track(ExtensionEvents.ConflictingExtensionsWarned, {
+      installedExtensions,
+    });
+    vscode.window
+      .showWarningMessage(
+        "We have detected some extensions that may conflict with Dendron. Further action is needed for Dendron to function correctly",
+        "Fix conflicts..."
+      )
+      .then(async (resp) => {
+        if (resp === "Fix conflicts...") {
+          await showConflictingExtensionPreview(installedExtensions);
+        }
+      });
+  }
+}
+
+async function showConflictingExtensionPreview(installedExtensions: string[]) {
+  const md = _md();
```

Comment:
instead of using `markdown-it`, the dendron preview now supports rendering arbitrary markdown. wrote a guide for it here -> [[Using the Dendron Preview on arbitrary markdown|dendron://dendron.docs/pkg.plugin-core.dev.cook#using-the-dendron-preview-on-arbitrary-markdown]]

one thing to check is if this has some activation requirements (eg. workspace needs to be activated) before this works. if there are no conflicts, we should prefer using this going forward

Reply:
`NoteUtils.createForFile` looks like it's used to open files that are not in a location that Dendron recognizes. (e.g. a markdown file that's outside a vault)

It takes a file path so I assume this is meant for an actual file that exists, just not in a vault.

Are you suggesting we should create a file with the content we want to show in the preview, and then show it?

Reply:
ah, `NoteUtils.createForFile` creates a dummy note for a file. for this change, we can add `NoteUtils.createForText` to pass in a note

Reply:
I can git `NoteUtils.createForFile` a random path, and it will indeed create a `NoteProps` for me, but I don't know if we want to do that.